scripts/zones/Windurst_Woods/npcs/Ponono.lua

```diff
@@ -11,34 +14,57 @@ require("scripts/globals/status")
 
 function onTrade(player,npc,trade)
     local newRank = tradeTestItem(player,npc,trade,tpz.skill.CLOTHCRAFT)
-
+    local moralmanifest = player:getQuestStatus(OTHER_AREAS_LOG,tpz.quest.id.otherAreas.A_MORAL_MANIFEST)
+    local tradeGil = trade:getGil()
     if newRank ~= 0 then
         player:setSkillRank(tpz.skill.CLOTHCRAFT,newRank)
         player:startEvent(10012,0,0,0,0,newRank)
     end
+    if moralmanifest == QUEST_ACCEPTED and player:getCharVar("moral") == 2 then
+        if npcUtil.tradeHas(trade, {828, 830{"gil", 10000}}) then -- Trade Velvet Cloth, Rainbow Cloth and 10k
```

Comment:
this line errors, misplaced/missing comma _I think_

Reply:
Can you give an example of npctil with items and Gil? Tried to find but couldn't.

Reply:
I think what you have is right, but as Teo suggested, you have a comma missing
```if npcUtil.tradeHas(trade, {828, 830,{"gil", 10000}}) then```
it should be like that i belive.

Reply:
let us know if that worked

Reply:
This is why testing again after you change something is important. If you have difficulty getting it to work when testing, hit folks up for help - discord is usually pretty active with ppl willing to assist

Reply:
@TeoTwawki  I usually do - Not sure why I didn't catch it... might have not saved it. I will ensure, moving forward, I double check.

Reply:
So I just ran through it entirely again and didn't have any hang ups. ( I did do give myself item 15202 and set moralWait to 0)